### PR TITLE
Update dskit version

### PR DESCRIPTION
**Pull request by MichelHollands**
**What this PR does / why we need it**:

Update the dskit version and update the Loki code so the new extra logger parameter for the RegisterFlags function of the LifecyclerConfig is given when registering the flags of the ingester.

This latest dskit version also updates the LifecyclerConfig so that only [private interfaces are used](https://github.com/grafana/dskit/pull/126). This required changes to unit tests. Additionally the private interface call was added to other rings: frontend, ruler and distributor. These do not use the LifeCycler from dskit.

The CircleCI config was updated as well because dskit now requires Go 1.17.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [X] Tests updated
- [x] Add an entry in the `CHANGELOG.md` about the changes.

Signed-off-by: Michel Hollands <michel.hollands@grafana.com>

### CODE REVIEW

**Comment by dannykopping** [comment]
> with the 1 impact

Could you elaborate on this please @MichelHollands?

**Comment by MichelHollands** [comment]
> > with the 1 impact
> 
> Could you elaborate on this please @MichelHollands?

The PR description has been updated so it mentions the extra parameter required by dskit in the ingester code.

**Review by dannykopping — COMMENTED** [review]
LGTM, ping me once you've got the tests passing and I'll approve + merge

File: vendor/github.com/grafana/dskit/ring/lifecycler.go
Comment (DylanGuedes) [inline vendor/github.com/grafana/dskit/ring/lifecycler.go:47]: do you know how one can use this new category feature?

**Review by DylanGuedes — APPROVED** [review]
LGTM! just a few minor questions

**Review by dannykopping — APPROVED** [review]
LGTM, I'm interested in your answers to @DylanGuedes' comments though before we merge

File: CHANGELOG.md
Comment (MichelHollands) [reply CHANGELOG.md:3]: Replaced with the etcd credentials message.

File: vendor/github.com/grafana/dskit/ring/lifecycler.go
Comment (MichelHollands) [reply vendor/github.com/grafana/dskit/ring/lifecycler.go:47]: I asked Christian who worked with the Mimir team. These tags are used by the doc-generator and the help messages to categorise settings. Loki/GEL have a separate doc-generator so the tags are ignored at the moment.

**Review by simonswine — COMMENTED** [review]
Thanks for pulling in this dskit improvement.

I think it would also make sense to override the `doc:` annotation:

```
InstanceInterfaceNames []string `yaml:"instance_interface_names" doc:"default=[<private network interfaces>]"
```

I have also noticed that you have not touched the `InstanceInterfaceNames` in `pkg/loki/common/common.go`

**Comment by MichelHollands** [comment]
> Thanks for pulling in this dskit improvement.
> 
> I think it would also make sense to override the `doc:` annotation in the same way that Mimir does it (https://github.com/grafana/mimir/pull/1165/files):
> 
> ```
> InstanceInterfaceNames []string `yaml:"instance_interface_names" doc:"default=[<private network interfaces>]"
> ```
> 
> I have also noticed that you have not touched the `InstanceInterfaceNames` in `pkg/loki/common/common.go`

The doc annotation has been added. 

Good catch about the common.go file. That has been updated as well now.

> Thanks for pulling in this dskit improvement.
> 
> I think it would also make sense to override the `doc:` annotation in the same way that Mimir does it (https://github.com/grafana/mimir/pull/1165/files):
> 
> ```
> InstanceInterfaceNames []string `yaml:"instance_interface_names" doc:"default=[<private network interfaces>]"
> ```
> 
> I have also noticed that you have not touched the `InstanceInterfaceNames` in `pkg/loki/common/common.go`

The annotations have been added.

> Thanks for pulling in this dskit improvement.
> 
> I think it would also make sense to override the `doc:` annotation in the same way that Mimir does it (https://github.com/grafana/mimir/pull/1165/files):
> 
> ```
> InstanceInterfaceNames []string `yaml:"instance_interface_names" doc:"default=[<private network interfaces>]"
> ```
> 
> I have also noticed that you have not touched the `InstanceInterfaceNames` in `pkg/loki/common/common.go`

@simonswine 
The doc annotation has been added. The documentation has been updated as well.
Good catch with common.go. That has been updated now as well.

**Review by simonswine — APPROVED** [review]
Thanks for addressing my comments. I think it's ready to go :rocket: . LGTM

**Pull request merged**
